### PR TITLE
Refactor `<ActivateUserModal />`

### PR DESCRIPTION
**Pull request by Andresbl123**
refactor the `<ActivateUserModal />` component (now `<ActivateFormOptions />`) so that it does not depend on the data structure of the form or assisted by `user`, but rather makes this data structure more generic for any form, where by means of `id` & `activate` you can do what you are looking for with the component in a more generic way,

This component must now be passed a generic, for TS help

### CODE REVIEW

**Comment by Andresbl123** [comment]
Tener en cuenta que por el momento se genera un nuevo error en cuanto a tipado, en el componente `<DecisionModal />`, ya que este tambien va ligado a la estructura de datos de `IGeneralInformationEntry` asaociados al formulario de usuarios, pienso que esto se debe resolver en una nueva PR, para que esta no se vuelve tan grande 
image

**Review by LinconOssa — APPROVED** [review]
Meets requirements

**Review by carin91 — CHANGES_REQUESTED** [review]
Verify several changes i made due to the merge of the main branch into this branch.

You should verify and approve these changes / adjust any elements that you considered.

File: src/pages/privileges/outlets/roles/config/dataRoles.tsx [hunk 1]
```diff
@@ -37,26 +38,44 @@ export const RolesBreakPointsConfig = [
...
-    Aplicación: roleselectd?.k_aplica,
-    Activo: roleselectd?.i_activo ? "Si" : "No",
-  }));
+const dataDetailsRol = (k_Rol: string) => {
+  const data = [mockRoles.find((role) => role.k_Rol === k_Rol)!].map(
+    (roleselectd) => ({
+      Código: roleselectd?.k_Rol,
+      Nombre: roleselectd?.n_Rol,
+      Aplicación: roleselectd?.n_Uso,
+      Activo: roleselectd?.i_Activo === "Y" ? "Si" : "No",
```
Comment (LinconOssa) [inline src/pages/privileges/outlets/roles/config/dataRoles.tsx:47]: I do not agree with working if and we cannot work with Active" : "Inactive"; or a better name

**Review by LinconOssa — COMMENTED** [review]
It is getting damaged in the fourth step it is getting undefined

File: src/pages/privileges/outlets/roles/config/dataRoles.tsx (hunk 1 above)
Comment (Andresbl123) [reply src/pages/privileges/outlets/roles/config/dataRoles.tsx]: resuelto

**Review by LinconOssa — APPROVED** [review]
Meets requirements

**Review by carin91 — APPROVED** [review]

**Pull request merged**